core/cont/src/TList.cxx

```diff
@@ -714,7 +714,7 @@ void TList::RecursiveRemove(TObject *obj)
    while (lnk) {
       next = lnk->Next();
       TObject *ob = lnk->GetObject();
-      if (ob->TestBit(kNotDeleted)) {
+      if (ob && ob->TestBit(kNotDeleted)) {
```

Comment:
... or given that *ob* is being created anyway
```
if (ob && ob->TestBit(kNotDeleted)  && ob->IsEqual(obj)) {
```
By the way, are we sure that all this is thread safe? All these operations happen in a finite time lapse.